web/crux/src/app/deploy/deploy.service.ts

```diff
@@ -886,6 +954,30 @@ export default class DeployService {
     })
   }
 
+  async onEditorJoined(
+    deploymentId: string,
+    clientToken: string,
+    identity: Identity,
+  ): Promise<[EditorMessage, EditorMessage[]]> {
+    const editors = await this.editorServices.getOrCreateService(deploymentId)
+
+    const me = editors.onClientJoin(clientToken, identity)
+
+    return [me, editors.getEditors()]
+  }
+
+  async onEditorLeft(deploymentId: string, clientToken: string): Promise<EditorLeftMessage> {
+    const editors = await this.editorServices.getOrCreateService(deploymentId)
+    const message = editors.onClientLeft(clientToken)
+
+    if (editors.editorCount < 1) {
+      this.logger.verbose(`All editors left removing ${deploymentId}`)
```

Comment:
What is the use case for this? I'm missing an understanding of when this situation might arise.

Reply:
For example two users open the same deployment to edit, then the editor service will have 2 editors for the same deployment, then both of them close their browsers, this way the deployment will have no editors, so we'll free the underlying services (which hold info about focused editors, editor names, colors, etc.).

Reply:
Okey, thanks.